docs/src/modules/components/AppSearch.js

```diff
@@ -153,8 +153,9 @@ const styles = theme => ({
     color: 'inherit',
   },
   inputInput: {
-    padding: `${theme.spacing.unit}px ${theme.spacing.unit}px ${theme.spacing.unit}px ${theme
-      .spacing.unit * 9}px`,
+    padding: `${theme.spacing(1)}px ${theme.spacing(1)}px ${theme.spacing(1)}px ${theme.spacing(
+      9,
+    )}px`,
```

Comment:
What do you think about accepting an array of values, and returning a string:

```
theme.spacing([1,2,4,1]) => '8px 16px 32px 8px'
```

Reply:
Related: https://github.com/mui-org/material-ui/pull/14099#discussion_r248200237

Reply:
I think this is already covered by `jss-expand` and/or `[1, 1, 1, 9].map(theme.spacing)`

Reply:
@joshwooding I might be misunderstanding, but that comment doesn't seem to be related. It seems to be about having an array of values in the theme one of which is returned according to the index.

@eps1lon Yeah, it looks like jss-expand with map would cover it. Not quite as elegant, but still workable. It would be good to document that (as well as use in the docs such as the above case).

It might still be nice to provide this natively if the overhead wasn't too high. It's a common enough scenario.

Reply:
This comes probably down to how much you "like" functional programming?
```diff
-theme.spacing([1, 2, 4, 1])
+[1, 2, 4, 1].map(theme.spacing)
```
is purely  syntactic sugar as far as I can tell. The first once requires additional implementation so I would argue that the second one is objectively better.

Reply:
👍 Let's document it then. (And use it for the docs!)

Reply:
@mbrookes my bad. I misunderstood the earlier comment

Reply:
We don't package *jss-plugin-expand* by default. People would have to install this jss plugin on their own. Why don't we have it? Because of https://bundlephobia.com/result?p=jss-plugin-expand@10.0.0-alpha.9 this: 1.4 kB gzipped. So the following doesn't work:
```js
[1, 2, 4, 1].map(theme.spacing)
```

I like the original suggestion of:
```js
theme.spacing(1, 2, 4, 1) => '8px 16px 32px 8px'
```
But it comes with some challenges:
- What happen if I provide a custom input function, e.g. `x => x ** 2`? Do I still get this feature?
- What if I want to use the `rem` unit?

Reply:
Ok. I have an idea.

Reply:
> What if I want to use the `rem` unit?

That's why I wouldn't be to opinionated about this. Functional programming makes this composable without being to verbose IMO:
`[1, 2, 4, 1].map(theme.spacing).map(asRem)`